editoast/src/views/mod.rs

```diff
@@ -105,14 +117,31 @@ async fn health(
     db_pool: Data<DbConnectionPoolV2>,
     redis_client: Data<RedisClient>,
 ) -> Result<&'static str> {
-    use diesel_async::RunQueryDsl;
-    sql_query("SELECT 1")
-        .execute(db_pool.get().await?.deref_mut())
-        .await?;
+    timeout(
+        Duration::from_millis(500),
+        check_health(db_pool.into_inner(), redis_client.into_inner()),
+    )
+    .await
+    .map_err(|_| AppHealthError::TimeoutError)??;
+    Ok("ok")
+}
 
+async fn check_health(
+    db_pool: Arc<DbConnectionPoolV2>,
+    redis_client: Arc<RedisClient>,
+) -> Result<()> {
+    let mut db_connection = db_pool.clone().get().await?;
+    let (_, _) = tokio::join!(
+        ping_database(db_connection.deref_mut()),
+        ping_redis(redis_client)
+    );
+    Ok(())
+}
+
+async fn ping_redis(redis_client: Arc<RedisClient>) -> Result<()> {
```

Comment:
Since you moved `ping_database` into `db_connection_pool.rs`, maybe this one can be moved to `redis_utils.rs`? I'd be fie to both have them here, or both in their respective module.